hooks/post-checkout

```diff
@@ -5,5 +5,6 @@ set -euo pipefail
 if [ "$BUILDKITE_PULL_REQUEST" == "false" ]; then
   pre-commit run -a
 else
+  git fetch origin "$BUILDKITE_PULL_REQUEST_BASE_BRANCH":"$BUILDKITE_PULL_REQUEST_BASE_BRANCH"
```

Comment:
Can you elaborate on why this is needed ? And how come only some back-ports fail?

Reply:
I saw this error a couple of times, this seems to impact backport PRs to release branches when we have to add new commits to the PRs (when we need to fix conflicts or fix some failing tests for example).

I was able to reproduce it locally and got the same error when I didn't have the target release branch locally and fetching the branch fixed the problem.

Reply:
Not sure if this is intermittent or happening all the time when we add another commit to the backport PR. I feel that it happened every time I had to add another commit to a backport PR but didn't really track it and this is such a rare case that I don't have a lot of occurence to measure it.

Reply:
This happened again in https://github.com/elastic/unified-release/pull/384 after I added a commit to the backport PR: https://buildkite.com/elastic/unified-release/builds/2379#018da399-cd93-4990-82bd-41d7e2bc902e

The error wasn't here on the backport PR before adding a new commit: https://buildkite.com/elastic/unified-release/builds/2378#018da392-f537-4316-946d-b189c60e9c53

I think that's enough to confirm this isn't intermittent or randomly but happening each time we have to add new commits on a backport PR.

Reply:
Another occurence in https://buildkite.com/elastic/unified-release/builds/2459#018dc1c4-8af9-41aa-bbda-e5b7213cd2ef